### ISSUE TITLE
Allow imports from directories above Meteor project root

### ISSUE THREAD

**Issue by clayne11** [meteor/meteor#8344]
My app is developed as a monorepo using [`lerna`](https://github.com/lerna/lerna). Lerna symlinks together all of the packages in the monorepo. It also detects when there are common dependencies between the modules and hoists them up to the repo root. This works fine for normal node applications, but it doesn't work with Meteor because Meteor doesn't look up the directory tree to find node modules.

Lerna improves bootstrapping time _massively_ for my app. It also fixes issues like having multiple versions of the same package (like React) which breaks React.

**Comment by ramezrafla** [comment]
Could you use symlinks within your meteor folder to point to where you include your packages?

**Comment by clayne11** [comment]
I could but that would be recreating a lot of the work that `lerna` is already doing in determining which dependencies to symlink and which ones can stay at a higher level. The other issue is that this is an incompatibility with the rest of the `npm` ecosystem which makes interoperability a pain.

**Comment by mjmasn** [comment]
Only just discovered this issue, it's going to cause problems for us. Are there any non-symlink workarounds?

**Comment by mjmasn** [comment]
@clayne11 agreed re compatibility with wider `npm` ecosystem, having to regularly context switch between Meteor and 'regular' Node isn't great fun. I believe there's value in stuff working as expected especially to Node developers who start using Meteor.

There's actually a couple of issues with the symlinking solution:

1. After symlinking I had to upgrade Meteor to fix #8423 (I think anything above 1.4.3.3-beta.4 works).

2. Given we're still lacking a `.meteorignore` solution the next issue was that Meteor tries to process all the files in the symlinked directory (including html coverage reports and large webpack builds of other apps etc.) and understandably runs out of memory. Creating multiple symlinks deeper into the folder structure _does work_ however we then have the issue that this symlinked directory isn't split into client/server so _(unless this has changed?)_ any change will reload the whole app rather than just e.g. client for client side changes. I can probably live with that for now, because the only other way to do this would be to individually symlink modules which doesn't seem very maintainable.

I also looked at Lerna but I think that's for published npm modules, whereas ours are just lightweight folders with a js file or two in them that can be imported.

Ideally I'd like to be able to specify relative imports that go above the app root, and either automatically or manually specify folders for meteor to watch. So if I have a git repo called `my-monorepo` that looks like this:
```
- my-monorepo/
    - apps/
        - meteor-app-1/
            - ... the app
    - builds/ (git ignored)
    - modules/
        - some-module1/
        - some-module2/
        - some-module3/
    - scripts/
        - ... some scripts
    - translations/
        - ... some json files
```

If I'm running meteor-app-1 I want to be able to watch the modules directory, ideally rebuilding client/server as required based on where the module was imported.

I'm probably describing Webpack actually...

**Comment by hwillson** [comment]
To help provide a more clear separation between feature requests and bugs, and to help clean up the feature request backlog, Meteor feature requests are now being managed under the https://github.com/meteor/meteor-feature-requests repository. 

This feature request will be closed here, but anyone interested in migrating this feature request to the new repository (to make sure it stays active), can click [here](https://github.com/meteor/meteor-feature-requests/issues/new?title=Allow%20imports%20from%20directories%20above%20Meteor%20project%20root&body=Migrated%20from:%20meteor/meteor%238344) to start the feature request migration process. This manual migration process is intended to help identify which of the older feature requests are still considered to be of value to the community. Thanks!
